src/chunk_list.cpp

```diff
@@ -9,34 +9,171 @@
 #include <cstring>
 #include <cstdlib>
 
+#include "uncrustify_types.h"
 #include "ListManager.h"
 #include "prototypes.h"
 #include "uncrustify.h"
 #include "space.h"
 
 
-typedef ListManager<chunk_t> ChunkList;
+typedef ListManager<chunk_t> ChunkList_t;
+
+/** use this enum to define in what direction or location an
+ * operation shall be performed. */
+enum loc_t
+{
+   before, /**< indicates a position or direction upwards (=pref) */
```

Comment:
I like the enum approach. I think the name it's a bit confusing though being in lowercase I thought it was a local variable. What do you say if we use Google naming style for enums, thus we have here `kBefore`, `kAfter` ?